packages/web-fragments/src/gateway/fragment-gateway.ts

```diff
@@ -54,10 +54,39 @@ export class FragmentGateway {
 		// create a reverse mapping of route patterns to fragment configs
 		// used for lookup when finding a route match.
 		fragmentConfig.routePatterns.forEach((routePattern) => {
-			const matcher = match(routePattern, {
+			const [pathnamePattern, searchPattern = ''] = routePattern.split('?');
+
+			const pathMatcher = match(pathnamePattern, {
 				decode: globalThis.decodeURIComponent,
 			});
 
+			const searchMatcher = (search: string) => {
+				// If there is no search param pattern rule in the route pattern, match the request
+				if (!searchPattern) return true;
+
+				// Compare the request search params with the search params in th route pattern
+				const searchParamsPattern = new URLSearchParams(searchPattern);
+				const searchParams = new URLSearchParams(search);
+
+				// For every search params pattern specified in the route config,
+				// Check against the decoded search param from the request.
+				// Only match the request if all search params match the pattern.
+				for (const [searchKey, searchValue] of searchParamsPattern) {
+					const matcher = match(searchValue, {
+						decode: globalThis.decodeURIComponent,
+					});
+
+					if (!matcher(searchParams.get(searchKey) || '')) return false;
+				}
+
+				return true;
+			};
+
+			const matcher = (urlPath: string) => {
```

Comment:
consider renaming this to `url` so that we don't have `urlPath` and `pathname` on the same line which is confusing.

Reply:
@rnguyen17 could you address this one please?